lmdeploy/pytorch/engine/model_agent.py

```diff
@@ -1247,6 +1254,25 @@ async def async_forward(self, inputs: ModelInputs, swap_in_map: SwapMap,
         return output
 
 
+# [Note] Exit By Sending Exit Flag
+# the registration to `atexit` of this function should be called
+# after importing torch.multiprocessing and the initialization of
+# distributed process group
+def _exit_by_sending_exit_flag(rank: int, agent: TPModelAgent):
+    if not hasattr(agent, 'stream'):
+        # agent is not initialized, just exits normally
+        return
+    # send exit_flag to all subprocess relying on all subprocess are alive
+    # and wait at _broadcast_inputs
+    exit_flag = True
+    _broadcast_inputs(rank, [None, None, None, exit_flag], agent.stream)
+    agent.stream.synchronize()
+    # Tricky, extra sleep for subprocess releasing resources
```

Comment:
Is it a MUST?

Reply:
> Is it a MUST?

After I test it in both ascend and cuda, the synchronization can be removed. I removed it, and we should wait for CI to find out whether this change could cause any other issue.